unit_tests/test_redis_aggregator.py

```diff
@@ -0,0 +1,334 @@
+# pylint: disable=missing-function-docstring, protected-access, no-member
+import json
+import uuid
+from unittest.mock import patch, PropertyMock, MagicMock, call
+import pytest
+from gsy_e_sdk.constants import LOCAL_REDIS_URL
+
+from gsy_e_sdk.redis_aggregator import RedisAggregator, RedisAggregatorAPIException
+
+TEST_AGGREGATOR_NAME = "TestAgg"
+
+TEST_TRANSACTION_ID = str(uuid.uuid4())
+TEST_AGGREGATOR_UUID = str(uuid.uuid4())
+
+TEST_DEVICE_UUID_1 = str(uuid.uuid4())
+TEST_DEVICE_UUID_2 = str(uuid.uuid4())
+
+TEST_BATCH_COMMAND_DICT = {
+    TEST_DEVICE_UUID_1: ["TEST_BATCH_COMMAND"],
+    TEST_DEVICE_UUID_2: ["TEST_BATCH_COMMAND"]
+}
+
+
+TEST_RESPONSE = {"status": "ready",
+                 "command": "offer",
+                 "offer": {"energy": 1,
+                           "price": 1,
+                           "seller": "test_seller"},
+                 "bid": {"energy": 1,
+                         "price": 1,
+                         "buyer": "test_buyer"},
+                 }
+
+TEST_MESSAGE_DICT = {
+    "data": {
+        "transaction_id": TEST_TRANSACTION_ID,
+        "status": "SELECTED",
+        "event": "market",
+        "aggregator_uuid": None,
+        "responses": {"TEST_ASSET_UUID": [TEST_RESPONSE]},
+        "device_uuid": None,
+        "grid_tree": {},
+        "trade_list": [{"trade_price": 1,
+                        "traded_energy": 1,
+                        "buyer": "anonymous",
+                        "seller": "test_seller"}],
+
+        "slot_completion": "0%",
+        "content": {
+            "slot_completion": "0%",
+            "start_time": "00:00",
+            "duration_min": 300,
+            "market_slot": 1, }
+    }
+}
+
+
+@pytest.fixture(autouse=True)
+def fixture_mock_connections(mocker):
+    mocker.patch("gsy_e_sdk.redis_aggregator.StrictRedis")
+    mocker.patch("gsy_e_sdk.redis_aggregator.ThreadPoolExecutor")
+    mocker.patch("gsy_e_sdk.redis_aggregator.Dict",
+                 return_value=TEST_MESSAGE_DICT)
```

Comment:
Is this still being used? Tests pass even when I comment this line out.

Reply:
I've confirmed that this line is apparently not necessary in order for tests to succeed. 
Originally, I thought I should mock the return value of this object as it is used as default _message_ for methods such as `_batch_response()`. I thought I would allow this method to execute (for it is a private method) as it should, but using the dictionary I forced it to use as a message instead of the one which lives in `typing` module. 
As this message Dict seem to hold information about the simulation, and transactions, I assumed it is connected to the redis server and gets updated after some process. However,  I do not know precisely how this is achieved, so my approach was to create a mocked Dict message inferred/reconstructed from the information the different methods request from this message.
In conclusion, I recreated this Dict message and mocked the actual Dict message used in the RedisAggregator in order to avoid a possible connection request with the server. The fact that even thought this lines are commented the tests succeed might imply this Dict does not make any kind of request to the server and therefore is not necessary to mock it.

Reply:
I'll address your comment by deleting these lines. However I would appreciate some insights in order to comprehend how these Dict messages work...

Reply:
I think there might be a misunderstanding in this case (please correct me if I didn't get your message!). In the line you mentioned:
```python
def _batch_response(self, message: Dict) -> None:
```
the `Dict` object is not the default value of `message` (in that case it would have been `message = Dict`, with the `=` sign in place of `:`).
Using `:` after a parameter we're defining the type of the parameter. So in this case we know that the `message` parameter is expected to be a dictionary. These are called "type hints" ([see docs](https://docs.python.org/3.8/library/typing.html)). Python is not strictly typed, so these types won't generate errors at runtime; but there are tools (like mypy) that automatically check if the types are respected. PyCharm should also have similar features built-in.

Reply:
You're right, I misread the `:` by `=` . Thank you for clearing my confusion!